app/services/jobs/return-logs/fetch-water-undertakers.service.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+/**
+ * Fetches all water undertakers that have a valid licence after 1/04/2025
+ * @module FetchWaterUndertakersService
+ */
+
+const { db } = require('../../../../db/db.js')
+const LicenceModel = require('../../../models/licence.model.js')
+const ReturnRequirementModel = require('../../../models/return-requirement.model.js')
+const ReturnVersionModel = require('../../../models/return-version.model.js')
+
+/**
+ * Fetches all water undertaker licences that will have not been ended before 1/04/2025
+ *
+ * @returns {Promise<module:LicenceModel>} the matching licences
+ */
+async function go() {
+  const quarterlyStartDate = new Date('2025-04-01')
+
+  return await LicenceModel.query()
+    .select([
+      'id',
+      'expiredDate',
+      'lapsedDate',
+      'licenceRef',
+      'revokedDate',
+      'startDate',
+      'waterUndertaker',
+      db.raw("regions->>'regionalChargeArea' as regionalChargeArea")
+    ])
+    .withGraphFetched('returnVersions')
+    .modifyGraph('returnVersions', (builder) => {
+      builder
+        .select(['id', 'startDate', 'reason'])
+        .where('startDate', '<', quarterlyStartDate)
+        .where('status', 'current')
+        // A return version must include return requirements in order for us to be able to copy from it
+        .whereExists(
+          ReturnRequirementModel.query()
+            .select(1)
+            .whereColumn('returnVersions.id', 'returnRequirements.returnVersionId')
+        )
+        .orderBy('startDate', 'desc')
+    })
+    .where('waterUndertaker', true)
+    .where((builder) => {
+      builder.whereNull('expiredDate').orWhere('expiredDate', '>=', quarterlyStartDate)
+    })
+    .where((builder) => {
+      builder.whereNull('lapsedDate').orWhere('lapsedDate', '>=', quarterlyStartDate)
+    })
+    .where((builder) => {
+      builder.whereNull('revokedDate').orWhere('revokedDate', '>=', quarterlyStartDate)
+    })
+    .whereNotExists(
+      ReturnVersionModel.query()
+        .where('returnVersions.status', 'current')
+        .where('quarterlyReturns', true)
```

Comment:
I am so sorry because I told you to put this there, and you'll have done a lot of testing. 😳 🤦 

The behaviour I intended was if the licence has a return version that starts on or after `2025-04-01`, then its because

- we've already run the migration
- a user has manually added the return version and determined quarterly is not needed

By leaving this clause in, we'll not handle the second scenario. If we take it out, we'll handle both.

So, again, very sorry!

```suggestion
```